textacy/spacier/doc_extensions.py

```diff
@@ -509,13 +516,17 @@ def to_bag_of_words(doc, normalize="lemma", weighting="count", as_strings=False)
     if as_strings is False:
         for wid, weight in wid_weights.items():
             lex = vocab[wid]
-            if not (lex.is_stop or lex.is_punct or lex.is_space):
+            if not ( (lex.is_stop and remove_stop) or 
```

Comment:
Do you know what the performance hit is on having these checks in the `for` loop? It's probably small, but I just want to be sure — I want to keep this function as fast as possible, since so many others depend on it.

Reply:
As far as I can see if the new flags are left at the default values, there very little performance difference compared to `textacy:master`:

**`chartbeat-labs/textacy:master`**

```
> import textacy
> cw = textacy.datasets.capitol_words.CapitolWords()
> cw.download()
> doc = textacy.make_spacy_doc(next(cw.texts()), lang='en')

> %%timeit -r 10 
words = textacy.spacier.doc_extensions.to_bag_of_words(doc, normalize="", 
                                                       as_strings=True)

34.6 µs ± 501 ns per loop (mean ± std. dev. of 10 runs, 10000 loops each)
```

**`kjoshi/textacy:master`**
```
> import textacy
> cw = textacy.datasets.capitol_words.CapitolWords()
> cw.download()
> doc = textacy.make_spacy_doc(next(cw.texts()), lang='en')

> %%timeit -r 10
words = textacy.spacier.doc_extensions.to_bag_of_words(doc, normalize="",
                                                       as_strings=True, 
                                                       filter_stops=True, 
                                                       filter_punct=True)
 
37.1 µs ± 799 ns per loop (mean ± std. dev. of 10 runs, 10000 loops each)
```

Of course, if the new filter flags are set to False, the performance is affected, since we're now keeping track of more tokens:
```
%%timeit -r 10
words = textacy.spacier.doc_extensions.to_bag_of_words(doc, normalize="",
                                                       as_strings=True, 
                                                       filter_stops=False, 
                                                       filter_punct=False)

45.6 µs ± 969 ns per loop (mean ± std. dev. of 10 runs, 10000 loops each)
```